lidofinance/ethereum-staking-widget
# feat: add the 'chain id toggler' instead of the 'chain type toggler'

### Description

- Migrate from 'chain type' to 'chain id' switcher.
- Add switchChain button in the Fallback banner
- Catch switch chain errors

### Demo

```
.env
SUPPORTED_CHAINS=1,10,17000,11155111,11155420
```

On the wrap page:

<img width="406" alt="Снимок экрана 2024-12-24 в 10 29 03" src="https://github.com/user-attachments/assets/67a4021c-f73c-42ba-86fd-6280327ed36f" />


On the Stake, WQ, Rewards, Settings pages (Testnet, for Prod the chain switcher is locked):

<img width="141" alt="image" src="https://github.com/user-attachments/assets/a0f6ba79-023c-405d-b2a8-ed610b6cc84e" />

The switchChain button in the Fallback banner (testnet widget demo):
On prod will be "Switch to Mainnet"

<img width="886" alt="Снимок экрана 2025-01-20 в 23 23 10" src="https://github.com/user-attachments/assets/904524c2-bdb4-4bd6-9082-71c21765ece6" />

Waiting the wallet (arrow is hidden)
<img width="589" alt="image" src="https://github.com/user-attachments/assets/f2315b34-921f-4430-8cdb-abfee5a9d683" />




### Code review notes

Pay attention that the 'wrap/unwrap' page has its own separate 'DappChainContext' context (the Stake, WQ, Rewards, Settings pages have a different common context). This is due to history.

### Testing notes

- L1 and L2 operations
- tx scan on modals
- change a chain on the wallet (a widget chain changes automatically)
- change a chain on the chain switcher (the wallet changes automatically or ask permissions)
- check the switchChain button in the Fallback banner
- select not allowed/not adde chain and
  - give permission/add network
  - wait 5000 seconds (you should see a console.warn)


### Checklist:

- [x] Checked the changes locally (MM and TW).
